deploy/chaos_crds.yaml

```diff
@@ -516,12 +516,117 @@ spec:
+                                    TODO: Add other useful fields. apiVersion,
+                                    kind, uid?'
+                                  type: string
+                                optional:
+                                  description: Specify whether the ConfigMap
+                                    or its key must be defined
+                                  type: boolean
+                              required:
+                                - key
+                              type: object
+                            fieldRef:
+                              description: 'Selects a field of the pod:
+                                supports metadata.name, metadata.namespace,
+                                metadata.labels, metadata.annotations, spec.nodeName,
+                                spec.serviceAccountName, status.hostIP,
+                                status.podIP.'
+                              properties:
+                                apiVersion:
+                                  description: Version of the schema the
+                                    FieldPath is written in terms of, defaults
+                                    to "v1".
+                                  type: string
+                                fieldPath:
+                                  description: Path of the field to select
+                                    in the specified API version.
+                                  type: string
+                              required:
+                                - fieldPath
+                              type: object
+                            resourceFieldRef:
+                              description: 'Selects a resource of the container:
+                                only resources limits and requests (limits.cpu,
+                                limits.memory, limits.ephemeral-storage,
+                                requests.cpu, requests.memory and requests.ephemeral-storage)
+                                are currently supported.'
+                              properties:
+                                containerName:
+                                  description: 'Container name: required
+                                    for volumes, optional for env vars'
+                                  type: string
+                                divisor:
+                                  anyOf:
+                                  - type: integer
+                                  - type: string
+                                  description: Specifies the output format
+                                    of the exposed resources, defaults to
+                                    "1"
+                                  pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
```

Comment:
Can we add a comment here to explain the significance of this regex pattern?

Reply:
this is standard naming constraint for divisor from kubernetes as explained in the properties description

Most environment variables in the Kubernetes API use the value field to carry simple values. However, the alternate valueFrom field allows you to specify a fieldRef to select fields from the pod’s definition, and a resourceFieldRef to select fields from one of its container’s definition.

The fieldRef field is a structure that has an apiVersion field and a fieldPath field. The fieldPath field is an expression designating a field of the pod. The apiVersion field is the version of the API schema that the fieldPath is written in terms of. If the apiVersion field is not specified it is defaulted to the API version of the enclosing object.

The fieldRef is evaluated and the resulting value is used as the value for the environment variable. This allows users to publish their pod’s name in any environment variable they want.

The resourceFieldRef is a structure that has a containerName field, a resource field, and a divisor field. The containerName is the name of a container, whose resource (cpu or memory) information is to be exposed. The containerName is optional for environment variables and defaults to the current container. The resource field is an expression designating a resource in a container, and the divisor field specifies an output format of the resource being exposed. If the divisor is not specified, it defaults to “1” for cpu and memory. The table shows possible values for cpu and memory resources for resource and divisor settings: